dispatcher/index-test.js

```diff
@@ -0,0 +1,568 @@
+'use strict';
+
+/*eslint-env mocha */
+/*eslint max-nested-callbacks: 0 */
+
+var sinon = require('sinon');
+var assert = require('assert');
+sinon.assert.expose(assert, {prefix: ''});
+
+var any = sinon.match.any;
+var same = sinon.match.same;
+var spy = sinon.spy;
+var stub = sinon.stub;
+
+var Promise = global.Promise || require('pinkie-promise');
+
+var Dispatcher = require('./index');
+var DataPointer = require('./data-pointer');
+var iteratorOf = require('../iterator-of');
+
+describe('dispatcher:', function() {
+  var dispatcher;
+  beforeEach(function() {
+    dispatcher = Dispatcher();
+  });
+
+  it('passes actions and data to all registered handlers', function() {
+    var handlers = [spy(), spy(), spy()];
+    handlers.forEach(function(handler) { dispatcher.register(handler); });
+
+    var action = {};
+    var data = {};
+    dispatcher.dispatch(action, DataPointer(data));
+
+    handlers.forEach(function(handler) {
+      assert.calledWith(handler, same(action), same(data));
+    });
+  });
+
+  describe('data subtrees:', function() {
+    var data;
+    beforeEach(function() {
+      data = {a: {}, b: [null, null, {c: {}}]};
+    });
+
+    it('passes subtrees of the data to handlers, depending on registration', function() {
```

Comment:
Interesting feature. Wondering what benefits you get from this in comparison to forwarding the full data tree?